lib/media/drm_engine.js

```diff
@@ -1341,31 +1397,34 @@ shaka.media.DrmEngine = class {
       // on Edge:  26 1d 5a 6e - 57 27 - d7 47 - 80 46 ea a5 d1 d3 4b 5a
       // Bug filed: https://bit.ly/2thuzXu
 
-      // NOTE that we skip this if byteLength != 16.  This is used for the IE11
-      // and Edge 12 EME polyfill, which uses single-byte dummy key IDs.
+      // NOTE that we skip this if byteLength != 16. This is used for the
+      // IE11 and Edge 12 EME polyfill, which uses single-byte dummy key IDs.
       // However, unlike Edge and Chromecast, Tizen doesn't have this problem.
-      if (this.currentDrmInfo_.keySystem == 'com.microsoft.playready' &&
-          keyId.byteLength == 16 &&
-          (shaka.util.Platform.isIE() || shaka.util.Platform.isEdge())) {
-        // Read out some fields in little-endian:
-        const dataView = shaka.util.BufferUtils.toDataView(keyId);
-        const part0 = dataView.getUint32(0, /* LE= */ true);
-        const part1 = dataView.getUint16(4, /* LE= */ true);
-        const part2 = dataView.getUint16(6, /* LE= */ true);
-        // Write it back in big-endian:
-        dataView.setUint32(0, part0, /* BE= */ false);
-        dataView.setUint16(4, part1, /* BE= */ false);
-        dataView.setUint16(6, part2, /* BE= */ false);
-      }
+      if (
+        shaka.media.DrmEngine.isPlayReadyKeySystem_(
+            this.currentDrmInfo_.keySystem
+        )) {
```

Comment:
So the new key system didn't even fix the key ID handling?  Is the only thing fixed robustness?

Reply:
The new key system seems to fix sessionType, persistentState, disctinctiveIdentifier and robustness handling (since they were just ignored in the previous keySytem)